Crashlytics/Crashlytics/Controllers/FIRCLSReportUploader.m

```diff
@@ -243,9 +198,67 @@ - (BOOL)uploadPackagedReportAtPath:(NSString *)path
                           asUrgent:(BOOL)urgent {
   FIRCLSDeveloperLog("Crashlytics:Crash:Reports", @"Submitting report%@",
                      urgent ? @" as urgent" : @"");
-  return [self submitPackageMultipartMimeAtPath:path
-                            dataCollectionToken:dataCollectionToken
-                                  synchronously:urgent];
+
+  // Check with the legacy path as the new path will always be contained in the legacy path
+  BOOL isNewPreparedPath = ![path containsString:self.fileManager.legacyPreparedPath];
+
+  if (isNewPreparedPath && self.dataSource.settings.shouldUseNewReportEndpoint) {
```

Comment:
Just a thought - should we just upload any reports that have been prepared, and just use the feature switch to determine how to prepare them?

Reply:
The app will have to fail to trigger an upload and the experiment flag has to be toggled to have reports in 2 prepared folder. Since this is an edge case, I am pivoting towards reliability. If there's issue with preparing or sending with one of the endpoints, then attempting to send to both endpoints may cause an unexpected issue (even crash). I prefer to have the SDK send only to the endpoint it is configured to.